modal/shared/protocol.py

```diff
@@ -8,6 +8,11 @@
 _COST_PER_SECOND_A100_80G: Final[float] = 0.001553
 
 
+class GPUType(Enum):
+    A100_40G = "A100_40G"
+    A100_80G = "A100_80G"
```

Comment:
was debating between having all gpu params bundled in this enum, so theres type+memory in the same unit

or breaking it out into type / memory separately (like how count is already a separate field?)

didn't wanna get too verbose with it tho, felt like a fine middle ground for now